resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/rest/kubernetes/v2/ResourceStagingServiceImplSuite.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.rest.kubernetes.v2
+
+import java.io.{ByteArrayInputStream, File}
+import java.nio.file.Paths
+
+import com.google.common.io.Files
+import org.scalatest.BeforeAndAfter
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.deploy.rest.KubernetesCredentials
+import org.apache.spark.util.Utils
+
+/**
+ * Unit, scala-level tests for KubernetesSparkDependencyServiceImpl. The coverage here
+ * differs from that of KubernetesSparkDependencyServerSuite as here we invoke the
+ * implementation methods directly as opposed to over HTTP, as well as check the
+ * data written to the underlying disk.
+ */
+class ResourceStagingServiceImplSuite extends SparkFunSuite with BeforeAndAfter {
```

Comment:
do you still need `with BeforeAndAfter` if you don't override `before` or `after` ?